gax-grpc/src/main/java/com/google/api/gax/grpc/ClientContext.java

```diff
@@ -27,43 +27,101 @@
  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
-package com.google.api.gax.grpc;
+package com.google.api.gax.rpc;
 
+import com.google.api.core.ApiClock;
 import com.google.api.core.BetaApi;
+import com.google.api.core.CurrentMillisClock;
+import com.google.api.gax.core.BackgroundResource;
+import com.google.api.gax.core.ExecutorAsBackgroundResource;
+import com.google.api.gax.core.ExecutorProvider;
 import com.google.auth.Credentials;
 import com.google.auto.value.AutoValue;
-import io.grpc.Channel;
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.Executors;
 import java.util.concurrent.ScheduledExecutorService;
 import javax.annotation.Nullable;
 
 /**
- * Encapsulates client state, including channel, executor, and credentials.
+ * Encapsulates client state, including executor, credentials, and transport context.
  *
  * <p>Unlike {@link ClientSettings} which allows users to configure the client, {@code
  * ClientContext} is intended to be used in generated code. Most users will not need to use it.
  */
 @BetaApi
 @AutoValue
 public abstract class ClientContext {
-  public abstract Channel getChannel();
+
+  /**
+   * The objects that need to be closed in order to clean up the resources created in the process of
+   * creating this ClientContext. This will include the closeables from the transport context.
+   */
+  public abstract List<BackgroundResource> getBackgroundResources();
 
   public abstract ScheduledExecutorService getExecutor();
 
   @Nullable
   public abstract Credentials getCredentials();
 
+  public abstract TransportContext getTransportContext();
+
+  public abstract ApiClock getClock();
+
   public static Builder newBuilder() {
-    return new AutoValue_ClientContext.Builder();
+    return new AutoValue_ClientContext.Builder()
+        .setBackgroundResources(Collections.<BackgroundResource>emptyList())
+        .setExecutor(Executors.newScheduledThreadPool(0))
+        .setTransportContext(NullTransportContext.create())
+        .setClock(CurrentMillisClock.getDefaultClock());
```

Comment:
You're right, it should be `NanoClock`. This should be changed in `ClientSettings` also.